### PR TITLE
Added triangle drawing

### PR DESCRIPTION
**Pull request by bodorkos-adam**
* now drawing cubes with triangles (with crappy algorithm)
* fixed some TODO things left over from previous PRs
* did not fix camera look issues yet. I think it's upside down, so now when you press it actually goes right. will fix

### CODE REVIEW

File: course6/toyapp-adam/displaylib/color.h [hunk 1]
```diff
@@ -11,6 +11,24 @@ struct Color
 	Channel r;
 	Channel g;
 	Channel b;
+
+	Color operator*(float intensity)
```
Comment (maciekgajewski) [inline course6/toyapp-adam/displaylib/color.h:15]: Should this ber const?

File: course6/toyapp-adam/displaylib/sdlobject.cc [hunk 2]
```diff
@@ -13,7 +13,7 @@ SDLObject::SDLObject()
 {
 	if(gOjectCounter++ == 0)
 	{
-		int res = SDL_Init(SDL_INIT_VIDEO);
+		auto res = SDL_Init(SDL_INIT_VIDEO);
```
Comment (maciekgajewski) [inline course6/toyapp-adam/displaylib/sdlobject.cc:16]: what was wrong with int?

File: course6/toyapp-adam/displaylib/surface_2d.cc [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+			if(error < 0)
+			{
+				y += ystep;
+				error += dx;
+			}
+		}
+	}
+}
+
+// slow, naive algorithm. it has overdraw when triangles share edges!
```
Comment (maciekgajewski) [inline course6/toyapp-adam/displaylib/surface_2d.cc:91]: That's fine. Don't optimise at this stage.

File: course6/toyapp-adam/displaylib/surface_2d.cc [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+void Surface2D::DrawTriangle(Pixel a, Pixel b, Pixel c, Color color)
```
Comment (maciekgajewski) [inline course6/toyapp-adam/displaylib/surface_2d.cc:92]: You don't need to fill the triangle. It's OK to just draw the edges.

File: course6/toyapp-adam/math/vector.h [hunk 5]
```diff
@@ -42,6 +71,28 @@ struct Vector3D
 		z *= scalar;
 		return *this;
 	}
+
+	Vector3D Normalized() const
+	{
+		auto inverseLength = 1.0f / Length();
```
Comment (maciekgajewski) [inline course6/toyapp-adam/math/vector.h:77]: what if Length is 0?

**Review by maciekgajewski — APPROVED** [review]
pretty good!

File: course6/toyapp-adam/displaylib/sdlobject.cc (hunk 2 above)
Comment (bodorkos-adam) [reply course6/toyapp-adam/displaylib/sdlobject.cc:16]: you're right, nothing. I just went through the codebase a couple times and opted for consistency with the usage of auto

File: course6/toyapp-adam/displaylib/color.h (hunk 1 above)
Comment (bodorkos-adam) [reply course6/toyapp-adam/displaylib/color.h:15]: oops, yes!

File: course6/toyapp-adam/math/vector.h (hunk 5 above)
Comment (bodorkos-adam) [reply course6/toyapp-adam/math/vector.h:77]: Good question. Probably would be best to return the vector unchanged, imho.

**Pull request merged**